### PR TITLE
fixing logic when multiple annotations need to be repositioned

### PR DESCRIPTION
**Pull request by danbowles**

### CODE REVIEW

File: app/components/ChartAnnotations/index.js [hunk 1]
```diff
@@ -15,12 +15,21 @@ class ChartAnnotations extends Component {
...
+    ChartAnnotations.updateAnnotation({
+      ...data,
+      ...ChartAnnotations.constrainAnnotation(data, container),
+    });
   }
 
   // Ensure annotations remain inside the chart container
   static constrainAnnotation(annotation, container) {
+    // Hide all but current annotation
+    d3v4.selectAll(`${container} .annotation`).each(function (data, index) {
```
Comment (meddy) [inline app/components/ChartAnnotations/index.js]: Why are you not using fat arrow function syntax?

File: app/components/ChartAnnotations/index.js [hunk 2]
```diff
@@ -190,6 +209,9 @@ class ChartAnnotations extends Component {
     this.setState({
       showAnnotations: !this.state.showAnnotations,
     });
+    if (this.state.showAnnotations) {
+      setTimeout((() => this.beginRenderingAnnotations()));
```
Comment (meddy) [inline app/components/ChartAnnotations/index.js]: Why do you need set timeout?

File: app/components/ChartAnnotations/index.js (hunk 1 above)
Comment (danbowles) [reply app/components/ChartAnnotations/index.js]: maintain 'this' for d3 - added comment.

File: app/components/ChartAnnotations/index.js (hunk 2 above)
Comment (danbowles) [reply app/components/ChartAnnotations/index.js]: ..thought i did.  guess not.  removed :)

**Pull request merged**